filters/openpolicyagent/openpolicyagent.go

```diff
@@ -573,27 +590,45 @@ func (m *bufferedBodyReader) Close() error {
 	return err
 }
 
-func (opa *OpenPolicyAgentInstance) ExtractHttpBodyOptionally(req *http.Request) (io.ReadCloser, []byte, error) {
+func bodyUpperBound(contentLength, maxBodyBytes int64) int64 {
+	if contentLength <= 0 {
+		return maxBodyBytes
```

Comment:
Do you have a test that has no body in the request?

Reply:
Actually no. Will add one...

Reply:
Added a few tests against unknown and nil bodies.